src/main/java/edu/cornell/kfs/gl/batch/service/impl/CuPosterServiceImpl.java

```diff
@@ -52,4 +60,161 @@ protected KualiDecimal getPercentage(KualiDecimal amount, BigDecimal percent) {
+            e.setFinancialObjectCode(et.getObjectCode());
+            e.setFinancialSubObjectCode(et.getSubObjectCode());
+        } else {
+            e.setFinancialObjectCode(icrRateDetail.getFinancialObjectCode());
+            e.setFinancialSubObjectCode(icrRateDetail.getFinancialSubObjectCode());
+        }
+
+        if (GeneralLedgerConstants.PosterService.SYMBOL_USE_EXPENDITURE_ENTRY.equals(icrRateDetail.getAccountNumber())) {
+            e.setAccountNumber(et.getAccountNumber());
+            e.setChartOfAccountsCode(et.getChartOfAccountsCode());
+            e.setSubAccountNumber(et.getSubAccountNumber());
+        } else if (GeneralLedgerConstants.PosterService.SYMBOL_USE_ICR_FROM_ACCOUNT.equals(icrRateDetail.getAccountNumber())) {
+            e.setAccountNumber(icrGenerationMetadata.getIndirectCostRecoveryAcctNbr());
+            e.setChartOfAccountsCode(icrGenerationMetadata.getIndirectCostRcvyFinCoaCode());
+            e.setSubAccountNumber(KFSConstants.getDashSubAccountNumber());
+        } else {
+            e.setAccountNumber(icrRateDetail.getAccountNumber());
+            e.setSubAccountNumber(icrRateDetail.getSubAccountNumber());
+            e.setChartOfAccountsCode(icrRateDetail.getChartOfAccountsCode());
+            // TODO Reporting thing line 1946
+        }
+        // take care of infinite recursive error case - do not generate entries
+        if ((et.getAccountNumber().equals(e.getAccountNumber())) &&
+            (et.getChartOfAccountsCode().equals(e.getChartOfAccountsCode())) &&
+            (et.getSubAccountNumber().equals(e.getSubAccountNumber())) &&
+            (et.getObjectCode().equals(e.getFinancialObjectCode())) &&
+            (et.getSubObjectCode().equals(e.getFinancialSubObjectCode()))) {
+            List<Message> warnings = new ArrayList<Message>();
+            warnings.add(new Message("Infinite recursive encumbrance error " + et.getChartOfAccountsCode() + " " + et.getAccountNumber() + " " + et.getSubAccountNumber() + " " + et.getObjectCode() + " " + et.getSubObjectCode(), Message.TYPE_WARNING));
+            reportWriterService.writeError(et, warnings);
+            return;
+        }
+
+        e.setFinancialDocumentTypeCode(parameterService.getParameterValueAsString(PosterIndirectCostRecoveryEntriesStep.class, KFSConstants.SystemGroupParameterNames.GL_INDIRECT_COST_RECOVERY));
+        e.setFinancialSystemOriginationCode(parameterService.getParameterValueAsString(KfsParameterConstants.GENERAL_LEDGER_BATCH.class, KFSConstants.SystemGroupParameterNames.GL_ORIGINATION_CODE));
+        SimpleDateFormat sdf = new SimpleDateFormat(DATE_FORMAT_STRING);
+        /*CUMod - start*/
+        StringBuffer docNbr = new StringBuffer(ICR_EDOC_PREFIX);
+        docNbr.append(sdf.format(runDate));
+        e.setDocumentNumber(docNbr.toString());
+        LOG.debug("CuPosterServiceImpl.generateTransactions: setDocumentNumber=" + docNbr.toString() + "=");
+        //e.setDocumentNumber(sdf.format(runDate));
```

Comment:
I think this comment could be removed, since your other comments already highlight the start and stop of the mod so that others know where to look to see the original code.

Reply:
I'll remove the original line of code.